src/common/location.rs

```diff
@@ -28,6 +29,18 @@ derive_header! {
     name: LOCATION
 }
 
+impl Location {
+    /// Creates a `Location` header from a uri
+    pub fn uri(uri: Uri) -> Self {
```

Comment:
I kinda wonder if we would want this function name as an accessor at some point... Not sure.

A `From` impl would work too.